src/compiler/checker.ts

```diff
@@ -1631,10 +1631,10 @@ namespace ts {
                 }
 
                 // If we're in an external module, we can't reference value symbols created from UMD export declarations
-                if (result && isInExternalModule && (meaning & SymbolFlags.Value) === SymbolFlags.Value && !(originalLocation!.flags & NodeFlags.JSDoc)) {
+                if (!compilerOptions.allowUmdGlobalAccess && result && isInExternalModule && (meaning & SymbolFlags.Value) === SymbolFlags.Value && !(originalLocation!.flags & NodeFlags.JSDoc)) {

```

Comment:
May be we should also report error if this flag is set true without UMD module resolution?(in verifyOptions in program.ts)

Reply:
If I understand correctly, @RyanCavanaugh’s comment on the test also applies here:

> The output/resolution format of the program doesn't really influence this flag - whether you import something from a UMD module is irrelevant to how you might import it